### PR TITLE
CAS-414 Reply view

### PR DESCRIPTION
**Pull request by bychkovdmitry**
https://stream-io.atlassian.net/browse/CAS-414

### Description

This PR contains `ReplyView` which will be embedded into every message view holder and configured via decorator.

### Checklist

- [X] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [X] PR targets the `develop` branch
- [ ] Changelog updated with client-facing changes
- [ ] New code is covered by unit tests
- [X] Comparison screenshots added for visual changes
- [X] Reviewers added

### CODE REVIEW

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/adapter/view/MessageReplyView.kt [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+    public constructor(context: Context) : super(context)
+    public constructor(context: Context, attrs: AttributeSet?) : super(context, attrs)
+    public constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr)
+
+    public fun setMessage(message: Message, isMine: Boolean) {
+        setUserAvatar(message)
+        setAvatarPosition(isMine)
+        setReplyBackground(message, isMine)
+        setAttachmentImage(message)
+        setReplyText(message)
```
Comment (zsmb13) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/adapter/view/MessageReplyView.kt:39]: Nice code organization 👍

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/adapter/view/MessageReplyView.kt [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    private fun setUserAvatar(message: Message) {
+        binding.avatarView.setUserData(message.user)
+    }
+
+    private fun setAvatarPosition(isMine: Boolean) {
+        binding.root.updateConstraints {
+            clear(R.id.avatarView, ConstraintSet.START)
+            clear(R.id.avatarView, ConstraintSet.END)
+            clear(R.id.replyContainer, ConstraintSet.END)
+            clear(R.id.replyContainer, ConstraintSet.END)
```
Comment (zsmb13) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/adapter/view/MessageReplyView.kt:51]: One of these is probably meant to clear the `START` constraint

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/adapter/view/MessageReplyView.kt [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+            .build()
+        val isLink = message.attachments
+            .lastOrNull()
+            ?.type == ModelType.attach_link
+        binding.replyContainer.background = MaterialShapeDrawable(shapeAppearanceModel).apply {
+            when {
+                isLink -> {
+                    paintStyle = Paint.Style.FILL
+                    setTint(context.getColorCompat(R.color.stream_ui_blue_alice))
+                }
```
Comment (zsmb13) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/adapter/view/MessageReplyView.kt:91]: Some of the design shows all attachments with this blue background (though some don't). Did we ask about the logic there?

Comment (bychkovdmitry) [reply stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/adapter/view/MessageReplyView.kt:91]: Asked designers, when original message contains link, then and only then we color background with light blue (The designs are already fixed in Figma).

**Review by zsmb13 — APPROVED** [review]

**Pull request merged**